frontend/src/services/api.ts

```diff
@@ -0,0 +1,46 @@
+import { getToken } from "./auth";
+import toast from "#/utils/toast";
+
+const WAIT_FOR_AUTH_DELAY_MS = 500;
+
+export async function request(
+  url: string,
+  optionsIn: RequestInit = {},
+  disableToast: boolean = false,
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+): Promise<any> {
+  const options = JSON.parse(JSON.stringify(optionsIn));
```

Comment:
nit

```suggestion
  const options = { ...optionsIn };
```

Reply:
IIUC this is a shallow copy, not a deep copy. And I think the linter complains about it :/

Reply:
Aha yeah it’s a shallow copy, which I thought is enough here… but never mind if linter is complaining! Shallow copies could lead to bugs sometimes.